frontends/p4/strengthReduction.cpp

```diff
@@ -80,7 +80,8 @@ const IR::Node *DoStrengthReduction::postorder(IR::BAnd *expr) {
     if (isAllOnes(expr->right)) return expr->left;
     auto l = expr->left->to<IR::Cmpl>();
     auto r = expr->right->to<IR::Cmpl>();
-    if (l && r) return new IR::Cmpl(new IR::BOr(expr->srcInfo, l->expr, r->expr));
+    if (l && r)
```

Comment:
The field `type` of expressions very rarely has the correct value.
But I guess there is no harm copying a wrong value.

Reply:
Assume you ran TypeChecking before applying this pass, the type should be correct, no?

Reply:
Only if you pass a specific flag to the typeChecking pass.
The problem is that adding types to the IR nodes changes the nodes (which are immutable), and thus invalidates any maps you may have built.